### DISCUSSION TITLE
Group all versions of a monorepo to a single MR

### DISCUSSION THREAD

**Discussion by BreakBB** [renovatebot/renovate#32713] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Renovate just detected updates of [@dnd-kit](https://github.com/clauderic/dnd-kit/releases), but all packages use a different (major) version, even though they depend on each other (e.g. `@dnd-kit/modifiers v8.0.0` requires `@dnd-kit/core v6.1.0`).

[The monorepo preset](https://docs.renovatebot.com/presets-group/#groupdnd-kitmonorepo) doesn't change anything to this, as three different version updates are detected. We do use `"separateMultipleMajor": true` if that makes a difference there.

Is there a way to let renovate put all updates into a single Merge Request independent on the version?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
Try turning `"separateMultipleMajor": false`.

**Reply by rarkins** [reply]
That defaults to false already

**Reply by BreakBB** [reply]
Well we use `"separateMultipleMajor": true` (as stated above). But from what I see that can only be set on global level. We only want that for this specific group.

**Comment by rarkins** [comment]
The dnd kit monorepo rule is based on a source URL: https://docs.renovatebot.com/presets-monorepo/#monorepodnd-kit

Are all the packages you care about from that same monorepo?

And if so, what branch names are you seeing?

**Reply by BreakBB** [reply]
Yes, all packages are from the same monorepo. See the releases here: https://github.com/clauderic/dnd-kit/releases

Using a `REN-` prefix the branch names are:

- `REN-dnd-kit-monorepo` (updating `@dnd-kit/core` from `6.1.0` to `6.2.0`)
- `REN-major-8-dnd-kit-monorepo` (updating `@dnd-kit/modifiers` from `7.0.0` to `8.0.0`)
- `REN-major-9-dnd-kit-monorepo` (updating `@dnd-kit/sortable` from `8.0.0` to `9.0.0`)

**Reply by rarkins** [reply]
So definitely set separateMultipleMajor back to default false (at least for this monorepo) and please update us on the result. You're explicitly telling Renovate to separate by major version while in actuality you don't want that.

**Reply by BreakBB** [reply]
But I can't set `separateMultipleMajor` for a single monorepo/group, can I? 🤔

**Reply by rarkins** [reply]
You can, just like the preset you linked to:

```
{
  "packageRules": [
    {
      "extends": [
        "monorepo:dnd-kit"
      ],
      "separateMultipleMajor": false,
      "matchUpdateTypes": [
        "digest",
        "patch",
        "minor",
        "major"
      ]
    }
  ]
}
```

**Reply by rarkins** [reply]
There's a little bit of a race condition though around lookups and major/minor bucketing. I think the above works, but if not then you need to write a rule using packageName patterns

**Reply by BreakBB** [reply]
I see, thanks.

https://docs.renovatebot.com/renovate-schema.json does not list this property, that's why I thought that's an invalid usage of `separateMultipleMajor`.

**Reply by BreakBB** [reply, marked as answer]
Using `separateMultipleMajor` in combination with `matchUpdateTypes` is not allowed (renovate raised an error). That was easy to fix and the result is now as expected. Thanks for helping out 👍🏻 

This is the config I am using now, to combine all `@dnd-kit` updates into a single MR:

```json
    "packageRules": [
        {
            "extends": [
                "monorepo:dnd-kit"
            ],
            "groupName": "dnd-kit monorepo",
            "separateMultipleMajor": false,
            "separateMajorMinor": false
        },
    ]
```
